Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -130,21 +133,54 @@ - (BOOL)canPerformAction:(SEL)action withSender:(id)sender {
     return [super canPerformAction:action withSender:sender];
 }
 
-- (void)paste:(id)sender {
+- (void)paste:(id)sender
+{
     if ([self pasteboardHasPossibleAttachment]) {
         SignalAttachment *attachment = [SignalAttachment attachmentFromPasteboard];
         // Note: attachment might be nil or have an error at this point; that's fine.
         [self.textViewPasteDelegate didPasteAttachment:attachment];
         return;
     }
-    
+
     [super paste:sender];
 }
 
 @end
 
 #pragma mark -
 
+@protocol OWSMessagesToolbarContentDelegate <NSObject>
+
+- (void)voiceMemoGestureDidStart;
+
+- (void)voiceMemoGestureDidEnd;
+
+- (void)voiceMemoGestureDidCancel;
+
+- (void)voiceMemoGestureDidChange:(CGFloat)cancelAlpha;
+
+@end
+
+#pragma mark -
+
+@interface OWSMessagesToolbarContentView ()
+
+@property (nonatomic, weak) id<OWSMessagesToolbarContentDelegate> delegate;
+
+@property (nonatomic) BOOL shouldShowVoiceMemoButton;
+
+@property (nonatomic) UIButton *voiceMemoButton;
+
+@property (nonatomic) UIButton *sendButton;
```

Comment:
In practice, the three properties won't be nil before they're accessed, but since they're lazy populated, I'm adding nullable.